### PR TITLE
feat: added add new and edit feature

### PR DESCRIPTION
**Pull request by nerdyspook**
**Features Added:**
> Add new note
> Edit note

**Misc**
> added note context
> note reducers

### CODE REVIEW

File: src/components/AddNote/AddNote.jsx [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+              );
+        setTitle("");
+        setBody("");
+        dispatchNote({ type: "DISPLAY_NOTE_EDITOR", payload: { show: false } });
+    };
+
+    return (
+        <div className="newnote__container">
+            <div className="top">
+                {/* <div className="title">Add a new note</div> */}
```
Comment (KanekarSnehal) [inline src/components/AddNote/AddNote.jsx:56]: remove comments

File: src/components/AddNote/AddNote.scss [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+            font-size: var(--h2-font-size);
+            position: absolute;
+            right: -0.5rem;
+            top: -1.9rem;
+        }
+    }
+
+    .note__title,
+    .note__body {
+        // position: absolute;
```
Comment (KanekarSnehal) [inline src/components/AddNote/AddNote.scss:14]: remove comments

File: src/Pages/Notes/Notes.jsx [hunk 3]
```diff
@@ -5,25 +5,48 @@ import Sidebar from "../../components/Sidebar/Sidebar";
...
-        </div>
+
+            <div className={`modal ${filter ? "" : "hidden"}`}>
+                <div className="modal-details">
+                    {filter && <Filter setFilter={setFilter} />}
+                </div>
+            </div>
+            <div className="modal-back hidden"></div>
+
+            <div className={`modal ${displayNoteEditor ? "" : "hidden"}`}>
```
Comment (KanekarSnehal) [inline src/Pages/Notes/Notes.jsx:43]: && can be used here

**Review by KanekarSnehal — APPROVED** [review]
Code is well written
Liked the use of Object.keys
Liked the use of useEffect for loading notes only once
No issues found other than few comments, code is good to merge.

**Review by Manpreet-hub — APPROVED** [review]
Code is clean & well structured. 
Liked the use of Objectkeys.

**Pull request merged**